CIP: Lower Sampling and Pruning Window to 14 days (#280)
celestiaorg/CIPs: cips/cip-sampling-window.md

> | cip | 33 |

Reviewer: ```suggestion
| cip | 34 |
```

big oops

> | cip | XX (assigned by Editors) |

Reviewer: ```suggestion
| cip | 33 |
```

| - | - |
| title | Lower Sampling and Pruning Window to 14 days |
| description | Lower Sampling and Pruning Window to 14 days |
| author | Callum Waters ([@cmwaters](https://github.com/cmwaters)), Nashqueue ([@Nashqueue](https://github.com/Nashqueue)) |

> | discussions-to | URL |

Reviewer: still needs discussion URL @cmwaters

Author: Added the link to the forum

| cip | XX (assigned by Editors) |
| - | - |
| title | Lower Sampling and Pruning Window to 14 days |
| description | Lower Sampling and Pruning Window to 14 days |
| author | Callum Waters ([@cmwaters](https://github.com/cmwaters)), Nashqueue ([@Nashqueue](https://github.com/Nashqueue)) |
| discussions-to | URL |
| status | Draft |
| type | Standards Track |
| category | Data Availability |
| created | 2025-03-17 |
| supercedes | [CIP 4](./cip-4.md) |

## Abstract

> Lower the minimum sampling and pruning window to 14 days and 14 days + 1 hour respectively to reduce storage costs in response to increasing throughput of the network and to ensure that the subjective intialized header, known as **Tail**, in the header pruning CIP, is within the trusting period and thus not subject to long range attacks.

Author: Like add it as a parameter

## Parameters

| Parameter     | Current value | Proposed value | Description  |
|---------------|---------------|----------------|--------------|
| RequestWindow | 720 hours  | 336 hours | This is analagous to the sampling window  |
| StorageWindow | 721 hours | 337 hours | This is analagous to the pruning window - the minimum retention period of shares for serving the network |

## Rationale

> The next pending increase in throughput at 32 MB blocks in 6 seconds could require as much as 1 TB of storage per day, at least 30TB of storage for pruned bridge nodes. With this CIP, it is reduced to 14TB making it less costly to serve data to the network. This may also promote the introduction of archival storage services or namespace pinning.

Reviewer 2: Kinda difficult to follow this logic. Here's an optional re-wording

```suggestion
The next pending increase in throughput (32 MB blocks every 6 seconds) would require bridge nodes to store up to 1 TB per day. The current sampling window (30 days) implies that pruned bridge nodes need to have 30 TB of storage. This CIP reduces the sampling window to 14 days to reduce the storage requirement to 14 TB which makes it less costly to serve data to the network. 
```

Also this seems unrelated to the original rationale and can be a separate paragraph:

> This may also promote the introduction of archival storage services or namespace pinning.

The current light node syncing logic involves performing sequential verification from genesis. Outside from the time and storage costs from syncing from genesis, it's also vulnerable to long range attacks as headers beyond the trusting period are based on validators with "nothing at stake". While the header pruning CIP, addresses these concerns, the safety issue is only addressed so long as the sampling window is equal to the trusting period (currently defaults as 14 days). Having a sampling window or header pruning window greater than the trusting period will compromise the security of a long range attack unless the node uses something like backwards sync.

> The one hour requires adequate tolerance to ensure that all nodes contains the headers needed to verify and sample the network.

Reviewer 2: Can this please be rephrased b/c it doesn't make sense to me. How does one hour require adequate tolerance? Maybe this meant to explain why StorageWindow = RequestWindow + 1?

Author: Yes it explains why there is 1 hour difference between the two parameters

## Backwards Compatibility

> Nodes should first lower the sampling window to 14 days and once the majority of the network has upgraded, lower the pruning window to 14 days + 1 hour.

Reviewer 3: By sampling window you mean the `RequestWindow` parameter?

Reviewer 2: 1. Why does one parameter have to be modified before the other?
2. Can we rename the parameters in code so that we only use one term for each concept? Preference for:

Rename `RequestWindow` to `SamplingWindow`
Rename `StorageWindow` to `PruningWindow?

Author: 1. I assume this is done in a rolling way which means DA nodes will upgrade at different times. If you modify both parameters in a single release, you may experience problems were nodes are trying to sample blocks that have already been pruned
2. Yeah totally agree. It was difficult to find these parameters

Reviewer: Agreed on 2 and that we should standardize these things across the protocol. Reminds me of seed nodes vs. bootstrappers too.
